### PR TITLE
feat: adds default scope extractor 

### LINKED ISSUE #961: IATP: default scope extractor
## WHAT

For each DSP request a set of minimum credentials will be required for consumer <-> provider communication.

Additional credentials will be derived by #834 

## WHY
iatp integration

## HOW

Implements a `DefaultScopeExtractor` as policy `pre-validator` that list the default required scopes.

### PR DESCRIPTION
**Pull request by wolf4ood**
## WHAT

This is a first IATP integration PR which contains:

The default scope extractor, which is the default credentials needed for DSP comunication e.g. `MembershipCredential`

In order to start testing IATP integration the `IatpHttpConsumerPullWithProxyInMemoryTest` has been created that
does a PULL scenario transfer with IATP presentation flow.

 The first IATP E2E test contains the following components:

- Two participants (connector) with their own credential service embedded.
- One remote STS minting the SI token for both participants.

The keys are are generated per participant and the STS and  CS service for simplicity they share the same private key.

The Membership credential for now has been created ahead of time and loaded in the CS store for the participants.

For this tests we used `did:example` as DID method with a custom cached `DIdResolver`

There is no policy functions yet that works on the credentials yet. Only `MembershipCredential` (default one) gets shared and validated.

## WHY

iatp integration

Closes #961

### CODE REVIEW

File: edc-extensions/iatp/tx-iatp/src/main/java/org/eclipse/tractusx/edc/iam/iatp/IatpScopeExtension.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
+/*
+ *  Copyright (c) 2023 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
```
Comment (paullatzelsperger) [inline edc-extensions/iatp/tx-iatp/src/main/java/org/eclipse/tractusx/edc/iam/iatp/IatpScopeExtension.java:2]:
```suggestion
 *  Copyright (c) 2024 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
```

File: edc-extensions/iatp/tx-iatp/src/main/java/org/eclipse/tractusx/edc/iam/iatp/scope/DefaultScopeExtractor.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
+/*
+ *  Copyright (c) 2023 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
```
Comment (paullatzelsperger) [inline edc-extensions/iatp/tx-iatp/src/main/java/org/eclipse/tractusx/edc/iam/iatp/scope/DefaultScopeExtractor.java:2]:
```suggestion
 *  Copyright (c) 2024 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
```

File: edc-extensions/iatp/tx-iatp/src/main/resources/META-INF/services/org.eclipse.edc.spi.system.ServiceExtension [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#
+#  Copyright (c) 2023 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
```
Comment (paullatzelsperger) [inline edc-extensions/iatp/tx-iatp/src/main/resources/META-INF/services/org.eclipse.edc.spi.system.ServiceExtension:2]:
```suggestion
#  Copyright (c) 2024 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
```

File: edc-extensions/iatp/tx-iatp/src/main/java/org/eclipse/tractusx/edc/iam/iatp/IatpScopeExtension.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+import static org.eclipse.edc.iam.identitytrust.core.IatpScopeExtractorExtension.CATALOG_REQUEST_SCOPE;
+import static org.eclipse.edc.iam.identitytrust.core.IatpScopeExtractorExtension.NEGOTIATION_REQUEST_SCOPE;
+import static org.eclipse.edc.iam.identitytrust.core.IatpScopeExtractorExtension.TRANSFER_PROCESS_REQUEST_SCOPE;
+import static org.eclipse.tractusx.edc.iam.iatp.IatpScopeExtension.NAME;
+
+@Extension(NAME)
+public class IatpScopeExtension implements ServiceExtension {
+    
+    static final String NAME = "Tractusx scope mapping extension";
+    static final Set<String> DEFAULT_SCOPES = Set.of("org.eclipse.tractusx.vc.type:MembershipCredential:read");
```
Comment (paullatzelsperger) [inline edc-extensions/iatp/tx-iatp/src/main/java/org/eclipse/tractusx/edc/iam/iatp/IatpScopeExtension.java:35]: i would make this configurable (and account for multiple default scopes), providing this as default value. 
also, the namespace (`org.eclipse.tractusx.vc.type`) should probably be configurable.

Comment (wolf4ood) [reply edc-extensions/iatp/tx-iatp/src/main/java/org/eclipse/tractusx/edc/iam/iatp/IatpScopeExtension.java:35]: we could do a list of scopes comma separated `org.eclipse.tractusx.vc.type:MembershipCredential:read,org.eclipse.tractusx.vc.type:SomeCredential:read` as config WDYT?

Comment (paullatzelsperger) [reply edc-extensions/iatp/tx-iatp/src/main/java/org/eclipse/tractusx/edc/iam/iatp/IatpScopeExtension.java:35]: comma-separated sounds good, but i would make all three parts configurable, i.e. the alias (`org.eclipse.tractusx.vc.type`), the type and the operation.

**Review by paullatzelsperger — APPROVED** [review]

**Pull request merged**